scripts/cleanOutdatedFrameworkProductions.py

```diff
@@ -107,60 +113,111 @@ def main(crabUsername, ingridUsername, DEBUG = False, evaluateSize = False):
         FWtag = tags[0]
         list_all_productions.append([FWtag, Anatag])
 
+    globalTotalSize = 0
+
+    # If needed, open output file
+    f = None
+    if output:
+        f = open(output, 'w')
+
     for i, p in enumerate(list_all_productions):
         if DEBUG and i > 0:
             break
         FWtag, Anatag = p
 
         extrastring = ''
         if not evaluateSize:
-            extrastring = '(evaluation of the disk size is OFF by default)'
-        print "\n##### Now looking at prod FWtag= ", FWtag, 'Anatag= ', Anatag, 'and list the associated folders %s' % extrastring
+            extrastring = ' (evaluation of the disk size is OFF by default)'
+        print("\n##### Now looking at prod FWtag: {}, Anatag: {} and list the associated folders{}".format(FWtag, Anatag, extrastring))
         totalSize = 0
-        totalSamples = 0
         cannotManageToDeleteThisProd = False
+
+        sample_ids = set()
+
         for s_name, s_id in list_allDBsamples:
             if FWtag in str(s_name) and Anatag in str(s_name):
                 result = dbstore.find(Sample, Sample.name == s_name)
                 s = result.one()
+
                 if evaluateSize:
                     totalSize += int(subprocess.check_output(["du", '-s', str(s.path)]).split()[0].decode('utf-8'))
-                totalSamples += 1
+
+                sample_ids.add(s.sample_id)
+
                 if s.source_sample is not None:
-                    print "WARNING, the sample", s_name, "depend on another sample, aborting now"
-                    cannotManageToDeleteThisProd = True
-                    break
-                if s.derived_samples.count() > 0:
-                    print "WARNING, the sample", s_name, "has derived samples, aborting now"
+                    print "WARNING, the sample", s.name, "depend on another sample, aborting now"
```

Comment:
I don't get this one...: below you're looking for derived samples, and add all those which have a source sample. So shouldn't just a `continue` be made here?